In [1]:
from transformers import AutoModelForCausalLM, AutoTokenizer
import torch

model_name = "Qwen/Qwen2.5-Math-1.5B"
# save_directory = "/data/a5-alignment/models/Qwen2.5-Math-1.5B"
save_directory = "../data/models/Qwen2.5-Math-1.5B"

model = AutoModelForCausalLM.from_pretrained(
    model_name,
    torch_dtype=torch.bfloat16,
    attn_implementation="flash_attention_2",
)
tokenizer = AutoTokenizer.from_pretrained(model_name)

You are attempting to use Flash Attention 2.0 with a model not initialized on GPU. Make sure to move the model to GPU after initializing it on CPU with `model.to('cuda')`.


generation_config.json:   0%|          | 0.00/138 [00:00<?, ?B/s]

tokenizer_config.json:   0%|          | 0.00/7.32k [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/2.78M [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/1.67M [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/7.03M [00:00<?, ?B/s]

In [2]:
model.save_pretrained(save_directory)
tokenizer.save_pretrained(save_directory)

('../data/models/Qwen2.5-Math-1.5B/tokenizer_config.json',
 '../data/models/Qwen2.5-Math-1.5B/special_tokens_map.json',
 '../data/models/Qwen2.5-Math-1.5B/vocab.json',
 '../data/models/Qwen2.5-Math-1.5B/merges.txt',
 '../data/models/Qwen2.5-Math-1.5B/added_tokens.json',
 '../data/models/Qwen2.5-Math-1.5B/tokenizer.json')

In [7]:
# Check if CUDA is available
if torch.cuda.is_available():
    model = model.to("cuda")
else:
    print("CUDA not available!")

In [8]:
prompt = "Solve for x: 5x + 10 = 35."
inputs = tokenizer(prompt, return_tensors="pt").to(model.device)

outputs = model.generate(**inputs, max_new_tokens=50)
result = tokenizer.decode(outputs[0], skip_special_tokens=True)

print(result)


Setting `pad_token_id` to `eos_token_id`:151643 for open-end generation.


Solve for x: 5x + 10 = 35. To solve the equation \(5x + 10 = 35\) for \(x\), we will follow these steps:

1. **Isolate the term with the variable \(x\)** by subtracting 10 from both sides


In [15]:
prompt = "Solve for x: 4x^2 - 10x + 35 = 0."
inputs = tokenizer(prompt, return_tensors="pt").to(model.device)

outputs = model.generate(**inputs, max_new_tokens=2000)
result = tokenizer.decode(outputs[0], skip_special_tokens=True)

print(result)

Setting `pad_token_id` to `eos_token_id`:151643 for open-end generation.


Solve for x: 4x^2 - 10x + 35 = 0. To solve the quadratic equation \(4x^2 - 10x + 35 = 0\), we will use the quadratic formula, which is given by:

\[
x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}
\]

Here, the coefficients \(a\), \(b\), and \(c\) are taken from the standard form of a quadratic equation \(ax^2 + bx + c = 0\). For our equation \(4x^2 - 10x + 35 = 0\), we have:

\[
a = 4, \quad b = -10, \quad c = 35
\]

First, we calculate the discriminant \(\Delta\), which is given by:

\[
\Delta = b^2 - 4ac
\]

Substituting the values of \(a\), \(b\), and \(c\):

\[
\Delta = (-10)^2 - 4 \cdot 4 \cdot 35
\]

\[
\Delta = 100 - 560
\]

\[
\Delta = -460
\]

Since the discriminant \(\Delta\) is negative, the quadratic equation has no real solutions. Instead, it has two complex solutions. We proceed by substituting \(\Delta\) back into the quadratic formula:

\[
x = \frac{-b \pm \sqrt{\Delta}}{2a}
\]

\[
x = \frac{-(-10) \pm \sqrt{-460}}{2 \cdot 4}
\]

\[
x = \frac{10 \pm \sqrt{-460}}{8}
\]

We can s

In [20]:
import regex as re

def extract_boxed_answer(text):
    match = re.search(r'\\boxed{((?:[^{}]+|{(?1)})*)}', text, re.DOTALL)
    return match.group(1).strip() if match else None

In [21]:
answer = extract_boxed_answer(result)
print(answer)

\frac{5}{4} + \frac{\sqrt{115}i}{4}, \frac{5}{4} - \frac{\sqrt{115}i}{4}


Solve for x: 4x^2 - 10x + 35 = 0. To solve the quadratic equation \(4x^2 - 10x + 35 = 0\), we will use the quadratic formula, which is given by:

\[
x = \frac{-b \pm \sqrt{b^2 - 4ac}}{2a}
\]

Here, the coefficients \(a\), \(b\), and \(c\) are taken from the standard form of a quadratic equation \(ax^2 + bx + c = 0\). For our equation \(4x^2 - 10x + 35 = 0\), we have:

\[
a = 4, \quad b = -10, \quad c = 35
\]

First, we calculate the discriminant \(\Delta\), which is given by:

\[
\Delta = b^2 - 4ac
\]

Substituting the values of \(a\), \(b\), and \(c\):

\[
\Delta = (-10)^2 - 4 \cdot 4 \cdot 35
\]

\[
\Delta = 100 - 560
\]

\[
\Delta = -460
\]

Since the discriminant \(\Delta\) is negative, the quadratic equation has no real solutions. Instead, it has two complex solutions. We proceed by substituting \(\Delta\) back into the quadratic formula:

\[
x = \frac{-b \pm \sqrt{\Delta}}{2a}
\]

\[
x = \frac{-(-10) \pm \sqrt{-460}}{2 \cdot 4}
\]

\[
x = \frac{10 \pm \sqrt{-460}}{8}
\]

We can simplify \(\sqrt{-460}\) using the imaginary unit \(i\), where \(i = \sqrt{-1}\):

\[
\sqrt{-460} = \sqrt{460} \cdot i
\]

\[
\sqrt{460} = \sqrt{4 \cdot 115} = 2\sqrt{115}
\]

Thus:

\[
\sqrt{-460} = 2\sqrt{115} \cdot i
\]

Substituting this back into the quadratic formula:

\[
x = \frac{10 \pm 2\sqrt{115}i}{8}
\]

We can simplify the fraction by dividing both the real and imaginary parts by 2:

\[
x = \frac{10}{8} \pm \frac{2\sqrt{115}i}{8}
\]

\[
x = \frac{5}{4} \pm \frac{\sqrt{115}i}{4}
\]

Therefore, the solutions to the quadratic equation \(4x^2 - 10x + 35 = 0\) are:

\[
x = \frac{5}{4} + \frac{\sqrt{115}i}{4} \quad \text{and} \quad x = \frac{5}{4} - \frac{\sqrt{115}i}{4}
\]

In box form, the solutions are:

\[
\boxed{\frac{5}{4} + \frac{\sqrt{115}i}{4}, \frac{5}{4} - \frac{\sqrt{115}i}{4}}
\]
